main.tf

```diff
@@ -1,65 +1,65 @@
-provider "aws" {
+  provider "aws" {
   region = "us-east-1"
 
 }
 
-resource "aws_instance" "app-server" {
-  instance_type = var.ec2-instance
+resource "aws_instance" "app-server" {          //zz problem, fix this by enabling kk for a resource with a module path: root             || for Non-attribute problems
+  instance_type = var.ec2-instance              //Change this to xx to save yy dollars annually for a resource with a module path: root      ||note: this var here in root is redundant & implies default value, otherwise vars are not going to get linted, only values
   ami           = "ami-09d56f8956ab235b3"
   tags = {
     created_for = "cloudfix-linter demo"
     Owner = "ankush.pandey@trilogy.com"
   }
-    root_block_device {
-    volume_type           = "gp2"
+    root_block_device {                          //zz problem, fix this by enabling kk for a resource with a module path: root  
+    volume_type           = "gp2"                //Change this to xx to save yy dollars annually for a resource with a module path: root
     volume_size           = "8"
     delete_on_termination = true
   }
-  ebs_block_device {
-    device_name           = "xvda"
-    volume_type           = "gp3"
+  ebs_block_device {                              //......
+    device_name           = "xvda"                
+    volume_type           = "gp3"                 //.......
     volume_size           = "8"
     delete_on_termination = true
   }
-      ebs_block_device {
-    device_name           = "xvdb"
-    volume_type           = "gp3"
+      ebs_block_device {                          //.......
+    device_name           = "xvdb"                
+    volume_type           = "gp3"                 //.......
     volume_size           = "8"
     delete_on_termination = true
   }
 
 }
 
-resource "aws_ebs_volume" "data-vol" {
+resource "aws_ebs_volume" "data-vol" {             //.......
   availability_zone = "us-east-1a"
   size              = 1
-  type              = "gp2"
+  type              = "gp2"                        //.......
   tags = {
     created_for = "cloudfix-linter demo"
     Owner = "ankush.pandey@trilogy.com"
   }
 }
 
-resource "aws_ebs_volume" "config-vol" {
+resource "aws_ebs_volume" "config-vol" {          //.......
   availability_zone = "us-east-1a"
   size              = 1
-  #type              = "gp2"
+  #type              = "gp2"                       //......
   tags = {
     created_for = "cloudfix-linter demo"
     Owner = "ankush.pandey@trilogy.com"
   }
 }
 
-module "auth" {
+module "auth" {                                // || Note: for modules blocks there are no global problems, only variable('var') problems are trasferred
   source            = ".//auth-module"
-  web_instance_type = "t2.micro"
-  ebs_device_type   = "gp2"
+  web_instance_type = "t2.micro"                // Change this to xx to save yy dollars for a resource with a module path: root->child->grandchild...   
```

Comment:
replace child->grandchild here with the exact values please.

Reply:
why are you leaving scope for mis-interpretation where it is so easily avoidable.